source/server/connection_handler_impl.cc

```diff
@@ -185,30 +185,41 @@ void ConnectionHandlerImpl::ActiveListener::onAccept(
 }
 
 void ConnectionHandlerImpl::ActiveListener::newConnection(Network::ConnectionSocketPtr&& socket) {
-  Network::ConnectionPtr new_connection = parent_.dispatcher_.createServerConnection(
-      std::move(socket), config_.transportSocketFactory().createTransportSocket());
+  // Find matching filter chain.
+  const auto filter_chain = config_.filterChainManager().findFilterChain(socket);
+  if (filter_chain == nullptr) {
+    ENVOY_LOG_TO_LOGGER(parent_.logger_, debug,
```

Comment:
Can you fix the build so that we can generate a coverage run?

Reply:
Sigh, sorry for the delay, I've got stuck trying to figure out why mocking `findFilterChain()` results in `socket` being destroyed... Ultimately, I gave up and implemented it in the tests instead.